File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Integer}
def longest_ones(nums, k)
  left = 0
  nums.each_index do |right|
      k -= 1 - nums[right]
      if k < 0
          k += 1 - nums[left]
          left+=1
      end
  end
  nums.size - left
end

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
   left=0;
   for (right in nums) {
      k-=1 - nums[right]
      if (k < 0) {
          k+=1 - nums[left]
          left+=1
      }
   }
   return nums.length - left
};